app/src/androidTest/java/com/github/sdp_begreen/begreen/firebase/FirebaseDBTest.kt

```diff
@@ -253,4 +253,192 @@ class FirebaseDBTest {
         }
     }
 
+    @Test
+    fun followThrowsIllegalArgumentExceptionWhenGivenBlankUserIdAsFirstArg() {
+        assertThrows(IllegalArgumentException::class.java) {
+            runBlocking {
+                FirebaseDB.follow("", "Non existent ID")
+            }
+        }
+    }
+
+    @Test
+    fun followDoesNothingWhenGivenNonExistentUserId() {
+        runBlocking {
+            FirebaseDB.follow("Non existent ID", "Non existent ID")
+        }
+    }
+
+    @Test
+    fun followCorrectlyAddsFollowerIdToFollowedUserAndFollowedIdToFollowerUser() {
+        val user1 = User("user1 id", 5)
+        val user2 = User("user2 id", 3)
+
+        // Add both users to the DB
+        runBlocking {
+            FirebaseDB.addUser(user1, user1.id)
+            FirebaseDB.addUser(user2, user2.id)
+        }
+
+        // Check that both users have no followers and following users
+        assertThat(user1.followers, nullValue())
+        assertThat(user1.following, nullValue())
+        assertThat(user2.followers, nullValue())
+        assertThat(user1.following, nullValue())
+
+        runBlocking {
+            FirebaseDB.follow(user1.id, user2.id)
+
+            val user1_modified = FirebaseDB.getUser(user1.id)!!
+            val user2_modified = FirebaseDB.getUser(user2.id)!!
+
+            // Check that the ids got added in the correct fields
+            assertThat(user1_modified.followers, nullValue())
+            assertThat(user1_modified.following, containsInAnyOrder(user2.id))
+            assertThat(user2_modified.followers, containsInAnyOrder(user1.id))
+            assertThat(user2_modified.following, nullValue())
+        }
+    }
+
+    @Test
+    fun unfollowThrowsIllegalArgumentExceptionWhenGivenBlankUserIdAsFirstArg() {
+        assertThrows(IllegalArgumentException::class.java) {
+            runBlocking {
+                FirebaseDB.unfollow("", "Non existent ID")
+            }
+        }
+    }
+
+    @Test
+    fun unfollowDoesNothingWhenGivenNonExistentUserId() {
+        runBlocking {
+            FirebaseDB.unfollow("Non existent ID", "Non existent ID")
+        }
+    }
+
+    @Test
+    fun unfollowCorrectlyRemovesFollowerIdFromFollowedUserAndFollowedIdFromFollowerUser() {
+        // user1 and user3 follow each other, user2 follows user1 and user3
+        var user1 = User("id1", 5,
```

Comment:
Another small suggestion: for better readability, you could add named arguments for id and score as well. Like : val user1 = User(id = "user1 id", score = 5, ....)

Reply:
I understand your point, but dont you have those strings displayed in your editor ?
<img width="143" alt="image" src="https://user-images.githubusercontent.com/71666958/236174908-bdd58538-956f-4c66-87a6-cec81c0e116e.png">

Reply:
Yes, I have it in my editor but not when I am reviewing the code directly on GitHub for example. But that is true that is not very important but it was more for consistency since some others have it.